tests/e2e/browser/utils.ts

```diff
@@ -0,0 +1,92 @@
+import { test, Page, Locator, expect } from "@playwright/test";
+import { getQuasarMenu } from "../locators";
+import {
+  mockReadFile,
+  mockShowOpenFileDialog,
+  mockShowSaveFileDialog,
+  mockWriteFile,
+} from "./mockUtility";
+import { UnreachableError } from "@/type/utility";
+
+/** UIのロックが解除されるまで待つ */
+export async function waitForUiUnlock(page: Page): Promise<void> {
+  await test.step("UIのロックが解除されるまで待つ", async () => {
+    const addAudioButton = page.getByLabel("テキストを追加");
+    await expect(addAudioButton).toBeEnabled({ timeout: 10 * 1000 });
+  });
+}
+
+/**
+ * ページ内でプロジェクトを読み込む。
+ *
+ * @param projectJson 読み込むプロジェクトファイルの内容
+ */
+export async function loadProject(
+  page: Page,
+  projectJson: string,
+): Promise<void> {
+  await test.step("プロジェクトを読み込む", async () => {
+    const testProjPath = `/tmp/${Date.now()}-testProj.vvproj`;
+    await mockReadFile(page, testProjPath, Buffer.from(projectJson, "utf-8"));
+    await mockShowOpenFileDialog(page, testProjPath);
+    await page.getByRole("button", { name: "ファイル" }).click();
+    await getQuasarMenu(page, "プロジェクトを読み込む").click();
+
+    // TODO: 保存ダイアログに対応する
+  });
+}
+
+/**
+ * プロジェクトを保存する。
+ *
+ * @returns 保存されたプロジェクトファイルの内容
+ */
+export async function saveProject(page: Page): Promise<string> {
+  return await test.step("プロジェクトを保存する", async () => {
+    const writeFileHandle = await mockWriteFile(page);
+    const saveFileDialogHandle = await mockShowSaveFileDialog(page);
+    await page.getByRole("button", { name: "ファイル" }).click();
+    await getQuasarMenu(page, "プロジェクトの複製を保存").click();
+    await waitForUiUnlock(page);
+    const [fileId] = await saveFileDialogHandle.getFileIds();
+    if (!fileId) {
+      throw new UnreachableError("assertion failed: fileId is undefined");
+    }
```

Comment:
こういう便利関数もあったり

https://github.com/VOICEVOX/voicevox/blob/5f1cced88ce30687f3ab91a8d79f355016b5f09f/src/type/utility.ts#L13-L20

```suggestion
    assertNonNullable(fileId)
```

まあエラーメッセージが分かりづらいけど。。